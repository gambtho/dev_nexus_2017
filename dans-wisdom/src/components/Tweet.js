import React from 'react'

const Tweet = () => ( <span/> )

export default Tweet

