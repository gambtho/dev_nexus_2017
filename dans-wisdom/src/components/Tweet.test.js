import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Tweet from './Tweet'

import tweets from '../twitter/sample-timeline.json'

const firstTweet = tweets[ 0 ]

describe( '<Tweet />', () => {

  it( 'renders without exploding', () => {
    const div = document.createElement( 'div' )
    ReactDOM.render( <Tweet />, div )
  } )

  it( `shows @dan_abramov's tweets`, () => {
		const el = renderer.create(<Tweet { ...firstTweet } />)
		expect(el.toJSON()).toMatchSnapshot()
  } )
} )
