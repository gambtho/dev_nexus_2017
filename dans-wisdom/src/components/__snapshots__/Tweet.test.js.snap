exports[`<Tweet /> shows @dan_abramov's tweets 1`] = `
<div
  className="tweet">
  <div
    className="context" />
  <div
    className="content">
    <div
      className="stream-item-header">
      <a
        className="account-group js-account-group js-action-profile js-user-profile-link js-nav"
        href="/dan_abramov">
        <img
          alt=""
          className="avatar js-action-profile-avatar"
          src="https://pbs.twimg.com/profile_images/826786122638426114/PR4tsq-i_normal.jpg" />
        <strong
          className="fullname js-action-profile-name show-popup-with-id">
          Dan Abramov
        </strong>
         
        <span>
          ‚Äè
        </span>
        <span
          className="username js-action-profile-name"
          data-aria-label-part="">
          <s>
            @
          </s>
          <b>
            dan_abramov
          </b>
        </span>
         
      </a>
       
      <small
        className="time">
         
        <a
          className="tweet-timestamp js-permalink js-nav js-tooltip"
          href="/dan_abramov/status/808735457785901057"
          title="10:08 AM - 13 Dec 2016">
          <span
            aria-hidden="true"
            className="_timestamp js-short-timestamp js-relative-timestamp"
            data-long-form="true"
            data-time="1481652533"
            data-time-ms="1481652533000">
            5h
          </span>
          <span
            className="u-hiddenVisually"
            data-aria-label-part="last">
            5 hours ago
          </span>
        </a>
      </small>
    </div>
    <div
      className="js-tweet-text-container">
      <p
        className="TweetTextSize js-tweet-text tweet-text"
        data-aria-label-part="0"
        lang="en">
        @ryanflorence @steida @nickhudkins @mjackson This is true (especially annoying you can\'t go \"back\" after regretting a nav transition)
      </p>
    </div>
    <div
      className="stream-item-footer">
      <div
        className="ProfileTweet-actionCountList u-hiddenVisually">
        <span
          className="ProfileTweet-action--reply u-hiddenVisually">
          <span
            className="ProfileTweet-actionCount"
            data-tweet-stat-count="6">
            <span
              className="ProfileTweet-actionCountForAria"
              data-aria-label-part="">
              0 replies
            </span>
          </span>
        </span>
        <span
          className="ProfileTweet-action--retweet u-hiddenVisually">
          <span
            aria-hidden="true"
            className="ProfileTweet-actionCount"
            data-tweet-stat-count="0">
            <span
              className="ProfileTweet-actionCountForAria">
              0 retweets
            </span>
          </span>
        </span>
        <span
          className="ProfileTweet-action--favorite u-hiddenVisually">
          <span
            className="ProfileTweet-actionCount"
            data-tweet-stat-count="16">
            <span
              className="ProfileTweet-actionCountForAria"
              data-aria-label-part="">
              1 likes
            </span>
          </span>
        </span>
      </div>
      <div
        aria-label="Tweet actions"
        className="ProfileTweet-actionList js-actions"
        role="group">
        <div
          className="ProfileTweet-action ProfileTweet-action--reply">
          <button
            className="ProfileTweet-actionButton u-textUserColorHover js-actionButton js-actionReply"
            data-modal="ProfileTweet-reply"
            type="button">
            <div
              className="IconContainer js-tooltip"
              title="Reply">
              <span
                className="Icon Icon--reply" />
              <span
                className="u-hiddenVisually">
                Reply
              </span>
            </div>
            <div
              className="IconTextContainer">
              <span
                className="ProfileTweet-actionCount ">
                <span
                  aria-hidden="true"
                  className="ProfileTweet-actionCountForPresentation" />
              </span>
            </div>
          </button>
        </div>
        <div
          className="ProfileTweet-action ProfileTweet-action--retweet js-toggleState js-toggleRt">
          <button
            className="ProfileTweet-actionButton  js-actionButton js-actionRetweet"
            data-modal="ProfileTweet-retweet"
            type="button">
            <div
              className="IconContainer js-tooltip"
              title="Retweet">
              <span
                className="Icon Icon--retweet" />
              <span
                className="u-hiddenVisually">
                Retweet
              </span>
            </div>
            <div
              className="IconTextContainer">
              <span
                className="ProfileTweet-actionCount ProfileTweet-actionCount--isZero">
                <span
                  aria-hidden="true"
                  className="ProfileTweet-actionCountForPresentation" />
              </span>
            </div>
          </button>
          <button
            className="ProfileTweet-actionButtonUndo js-actionButton js-actionRetweet"
            data-modal="ProfileTweet-retweet"
            type="button">
            <div
              className="IconContainer js-tooltip"
              title="Undo retweet">
              <span
                className="Icon Icon--retweet" />
              <span
                className="u-hiddenVisually">
                Retweeted
              </span>
            </div>
            <div
              className="IconTextContainer">
              <span
                className="ProfileTweet-actionCount ProfileTweet-actionCount--isZero">
                <span
                  aria-hidden="true"
                  className="ProfileTweet-actionCountForPresentation" />
              </span>
            </div>
          </button>
        </div>
        <div
          className="ProfileTweet-action ProfileTweet-action--favorite js-toggleState">
          <button
            className="ProfileTweet-actionButton js-actionButton js-actionFavorite"
            type="button">
            <div
              className="IconContainer js-tooltip"
              title="Like">
              <div
                className="HeartAnimationContainer">
                <div
                  className="HeartAnimation" />
              </div>
              <span
                className="u-hiddenVisually">
                Like
              </span>
            </div>
            <div
              className="IconTextContainer">
              <span
                className="ProfileTweet-actionCount">
                <span
                  aria-hidden="true"
                  className="ProfileTweet-actionCountForPresentation" />
              </span>
            </div>
          </button>
          <button
            className="ProfileTweet-actionButtonUndo u-linkClean js-actionButton js-actionFavorite"
            type="button">
            <div
              className="IconContainer js-tooltip"
              title="Undo like">
              <div
                className="HeartAnimationContainer">
                <div
                  className="HeartAnimation" />
              </div>
              <span
                className="u-hiddenVisually">
                Liked
              </span>
            </div>
            <div
              className="IconTextContainer">
              <span
                className="ProfileTweet-actionCount">
                <span
                  aria-hidden="true"
                  className="ProfileTweet-actionCountForPresentation" />
              </span>
            </div>
          </button>
        </div>
        <div
          className="ProfileTweet-action ProfileTweet-action--more js-more-ProfileTweet-actions">
          <div
            className="dropdown">
            <button
              aria-haspopup="true"
              className="ProfileTweet-actionButton u-textUserColorHover dropdown-toggle js-dropdown-toggle"
              type="button">
              <div
                className="IconContainer js-tooltip"
                title="More">
                <span
                  className="Icon Icon--dots" />
                <span
                  className="u-hiddenVisually">
                  More
                </span>
              </div>
            </button>
            <div
              className="dropdown-menu">
              <div
                className="dropdown-caret">
                <div
                  className="caret-outer" />
                <div
                  className="caret-inner" />
              </div>
              <ul>
                <li
                  className="copy-link-to-tweet js-actionCopyLinkToTweet">
                  <button
                    className="dropdown-link"
                    type="button">
                    Copy link to Tweet
                  </button>
                </li>
                <li
                  className="embed-link js-actionEmbedTweet"
                  data-nav="embed_tweet">
                  <button
                    className="dropdown-link"
                    type="button">
                    Embed Tweet
                  </button>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
